### ISSUE TITLE
Empty responses are not closed in HTTP/2

### ISSUE THREAD

**Issue by fanzeyi** [actix/actix-web#709]
This might be related to #439.

When the server is started with `.bind_ssl` and the handler returns an empty string, the connection won't be closed. (Update: I noticed it's probably a bug related to HTTP/2)

Here is a minimal code to reproduce this problem:

```rust
use actix_web::{http, server, App, Path, Responder};
use openssl::ssl::{SslAcceptor, SslFiletype, SslMethod};

fn index(_: Path<(u32, String)>) -> impl Responder {
    ""
}

fn main() {
    let mut builder = SslAcceptor::mozilla_intermediate(SslMethod::tls()).unwrap();
    builder
        .set_private_key_file("key.pem", SslFiletype::PEM)
        .unwrap();
    builder.set_certificate_chain_file("cert.pem").unwrap();

    server::new(|| App::new().route("/{id}/{name}/index.html", http::Method::GET, index))
        .bind_ssl("127.0.0.1:8080", builder)
        .unwrap()
        .run();
}
```

Cargo.toml:
```rust
[package]
name = "actix-empty-ssl"
version = "0.1.0"
authors = ["Zeyi Fan <zeyi@fb.com>"]
edition = "2018"

[dependencies]
actix-web = { version = "0.7.18", features = ["ssl"] }
openssl = "0.10.18"
```

```bash
$ curl -k -i https://localhost:8080/123/abc/index.html
HTTP/2 200
content-type: text/plain; charset=utf-8
date: Wed, 27 Feb 2019 22:30:04 GMT
content-length: 0

# cURL hangs here forever.
```

Update:

If I pass `--http1.1` to curl, the connection would finish properly:

```bash
$ curl -k --http1.1 -i https://localhost:8080/123/abc/index.html
HTTP/1.1 200 OK
content-length: 0
content-type: text/plain; charset=utf-8
date: Wed, 27 Feb 2019 22:36:46 GMT
```

**Comment by fanzeyi** [comment]
I tried to debug this issue and I think the bug might be inside the `poll_completed` function of H2Writer. The H2Writer is being polled forever with the flags `STARTED` and `EOF`, and because the response is empty (so the buffer is empty). The `reserve_capacity` function is never called. This causes poll_capacity keeps returning `Async::NotReady`. 

I attempted to fix this but I don't think this is correct, but I'm posting it here anyways:

```diff
diff --git a/src/server/h2writer.rs b/src/server/h2writer.rs
index fef6f88..0dff794 100644
--- a/src/server/h2writer.rs
+++ b/src/server/h2writer.rs
@@ -234,6 +234,16 @@ impl<H: 'static> Writer for H2Writer<H> {
                 stream.reserve_capacity(cmp::min(self.buffer.len(), CHUNK_SIZE));
             }

+            if self.flags.contains(Flags::EOF)
+                && !self.flags.contains(Flags::RESERVED)
+                && self.buffer.is_empty()
+            {
+                if let Err(e) = stream.send_data(Bytes::new(), true) {
+                    return Err(io::Error::new(io::ErrorKind::Other, e));
+                }
+                return Ok(Async::Ready(()));
+            }
+
             loop {
                 match stream.poll_capacity() {
                     Ok(Async::NotReady) => return Ok(Async::NotReady),
```

**Comment by fanzeyi** [comment]
I saw 1.0 is going to be released so I tried to use the 1.0 branch with this code. The empty response is now correctly returned. However, there is a high CPU usage issue; for every new request, the CPU usage of the process is increased by 100%. This bug is consistent no matter the response is empty or not. Opening another issue.

**Comment by fafhrd91** [comment]
lets keep it open for now, i'd like to fix 0.7

**Comment by fafhrd91** [comment]
fixed by #737
